### DISCUSSION TITLE
How to construct API with 20+ route handlers?

### DISCUSSION THREAD

**Discussion by jpaju** [sukovanej/effect-http#585] [Q&A] [answered]
Hi! Thanks for creating such awesome library for Typescript 🙏  

I have an backend API where I need to have 20 or more endpoints in a single server. I'm able to define the endpoints just fine, using a couple of `ApiGroup`s. However when trying to implement the endpoints by defining handlers for them as shown in the docs:
```typescript
const api = Api.make().pipe(Api.addGroup(apiGroup1), Api.addGroup(apiGroup2));

RouterBuilder.make(api).pipe(
  RouterBuilder.handle("group1-endpoint1", request => /* handler logic */),
  // ... other 18 handlers
  RouterBuilder.handle("group2-endpoint10", request => /* handler logic */),
  RouterBuilder.build
)
```

I'm unable to do so because `.pipe` only accepts 20 arguments and I get a compilation error `Expected 1-20 arguments, but got 21`.

Ideally I would just define handlers for each `ApiGroup` separately and then combine handlers/`RouterBuilder`s for each group, something like this (totally imaginary syntax/API in this example) :
```typescript
const routerBuilder1 = RouterBuilder.make(apiGroup1).pipe(
  RouterBuilder.handle("group1-endpoint1", request => /* handler logic */),
  // ...
)

const routerBuilder2 = RouterBuilder.make(apiGroup2).pipe(
  RouterBuilder.handle("group2-endpoint1", request => /* handler logic */),
  // ... 10 or so handlers
)

RouterBuilder.combine(routerBuilder1, routerBuilder2).pipe(RouterBuilder.build) // Something like this to compose all builders together
```

How could I circumvent this limitation and create handlers for over 20 routes?

**Comment by sukovanej** [comment, marked as answer]
Hey, good point 👍. I added `RouterBuilder.merge`.

```ts
const app = RouterBuilder.merge(routerBuilder1, routerBuilder2).pipe(RouterBuilder.build)

// or

const app = routerBuilder1.pipe(
  RouterBuilder.merge(routerBuilder2),
  RouterBuilder.merge(routerBuilder3),
  ...,
  RouterBuilder.build
)
```

You can use check the [tests](https://github.com/sukovanej/effect-http/blob/master/packages/effect-http-node/test/router-builder.test.ts#L414) or [docs](https://sukovanej.github.io/effect-http/effect-http/RouterBuilder.ts.html#merge)

**Comment by jpaju** [comment]
Awesome, thanks a lot! 🚀
